### ISSUE TITLE
JSHandle@node is not supported

### ISSUE THREAD

**Issue by scallaway** [argos-ci/jest-puppeteer#540]
I'm receiving the error in the title when trying to call `expect` on values that are returned from either `Page.waitForSelector()` or `Page.$()`.

I've had a look through the source and it looks as though `expect-puppeteer` only seems to be handling `ElementHandle` and not `JSHandle`, even though the former extends the latter.

Versions of the various packages;
```
    "jest": "^29.4.1",
    "jest-puppeteer": "^8.0.3",
    "puppeteer": "^19.6.1",
```

I can confirm that the `expect` call works if I explicitly import it from `jest`, but I'd much prefer to not need to do that throughout all our tests.

I don't believe this to be a bug (hence my not creating an issue of that type), but I would love to get some insight on the matter!

**Comment by gregberge** [comment]
Hello @scallaway, can you be more explicit on the error you got? TypeScript one?

**Comment by scallaway** [comment]
Hi @gregberge 

Sorry for the lack of information!

I don't believe it to be a TypeScript error, since it looks as though it's being thrown from the `expect-puppeteer` code itself.

The complete error is;
```
    JSHandle@node is not supported

       8 |
       9 | export function assertExists<T>(value: T): asserts value is NonNullable<T> {
    > 10 |   expect(value).not.toBeNull();
         |   ^
      11 |   expect(value).not.toBeUndefined();
      12 | }
      13 |

      at expectPuppeteerInstance (node_modules/expect-puppeteer/dist/index.js:482:11)
      at expectPuppeteer (node_modules/expect-puppeteer/dist/index.js:486:26)
```

`assertExists` is just a helper function that we have to keep the TS compiler happy when we're dealing with values that it believes _should_ be nullable, even though we can verify they're not (and the test should fail if they are).

The case I believe we're hitting is here: https://github.com/argos-ci/jest-puppeteer/blob/main/packages/expect-puppeteer/src/index.ts#L145

**Comment by gregberge** [comment]
@scallaway thanks, could you please give me the input provided? Because [Page.waitForSelector()](https://pptr.dev/api/puppeteer.page.waitforselector) and [Page.$()](https://pptr.dev/api/puppeteer.page._) returns ElementHandle according to the documentation.

**Comment by scallaway** [comment]
So if I log the value that's being passed to `assertExists` (that fails), I get

	value:  ElementHandle {}

which is strange because that should be handled. I'm not quite sure where the `JSHandle` stuff is coming from.

The relevant code is;

	const SELECTOR_FILTER_INPUT = `[aria-label="label filter"] [role="search"] input`;
	const inputField = await page.waitForSelector(SELECTOR_FILTER_INPUT);


It's also worth noting that I recently updated us from version 6 (where everything was working fine), to the latest where we're now seeing this problem.

**Comment by gregberge** [comment]
What Puppeteer version do you use? Could you please dig it more? Because according to the spec. of Puppeteer it is right, also all tests are passing here.

**Comment by scallaway** [comment]
So this was rather embarrassing.

I did a bit more digging, and your question about the `puppeteer` version that we're using got me thinking.

So I updated to the latest version of both `jest` and `puppeteer`, and that seemed to fix the problem that I was seeing. I have a feeling that it was some incompatibility that exists between certain versions of `expect-puppeteer` and `puppeteer`.

Thanks for being so responsive, and thanks for all the work that you're putting into this adapter!
